pulsar-client-go/pulsar/producer.go

```diff
@@ -166,13 +166,26 @@ type Producer interface {
 	// This call will be blocking until is successfully acknowledged by the Pulsar broker.
 	// Example:
 	// producer.Send(ctx, pulsar.ProducerMessage{ Payload: myPayload })
+	// @Deprecated
 	Send(context.Context, ProducerMessage) error
 
+	// Send a message
+	// This call will be blocking until is successfully acknowledged by the Pulsar broker.
+	// Example:
+	// msgID, err := producer.SendWithMsgID(ctx, pulsar.ProducerMessage{ Payload: myPayload })
+	SendWithMsgID(context.Context, ProducerMessage) (MessageID, error)
```

Comment:
From this method name, I get the impression that "Message ID is sent to broker with something" or "Something is sent to broker using message ID". What about a name like `SendAndGetMsgId`?